File: go-sdk/pkg/models/operations/createabranch.go
package operations

type CreateABranchPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type CreateABranchRequestBody struct {
	BackupID     *string `json:"backup_id,omitempty"`
	Name         string  `json:"name"`
	ParentBranch string  `json:"parent_branch"`
}

type CreateABranchRequest struct {
	PathParams CreateABranchPathParams
	Request    *CreateABranchRequestBody `request:"mediaType=application/json"`
}

type CreateABranchResponse struct {
	ContentType                           string
	StatusCode                            int64
	CreateABranch201ApplicationJSONObject map[string]map[string]interface{}
}

File: python-sdk/src/planetscaleapi/models/operations/create_a_branch.py
from __future__ import annotations
import dataclasses
from dataclasses_json import Undefined, dataclass_json
from planetscaleapi import utils
from typing import Any, Optional


@dataclasses.dataclass
class CreateABranchPathParams:
    database: str = dataclasses.field(metadata={'path_param': { 'field_name': 'database', 'style': 'simple', 'explode': False }})
    organization: str = dataclasses.field(metadata={'path_param': { 'field_name': 'organization', 'style': 'simple', 'explode': False }})
    

@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class CreateABranchRequestBody:
    name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    parent_branch: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('parent_branch') }})
    backup_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('backup_id'), 'exclude': lambda f: f is None }})
    

@dataclasses.dataclass
class CreateABranchRequest:
    path_params: CreateABranchPathParams = dataclasses.field()
    request: Optional[CreateABranchRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class CreateABranchResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    create_a_branch_201_application_json_object: Optional[dict[str, dict[str, Any]]] = dataclasses.field(default=None)
    